src/compiler/checker.ts

```diff
@@ -26217,6 +26218,10 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
         }
 
         function inferToTemplateLiteralType(source: Type, target: TemplateLiteralType) {
+            if (isTemplateLiteralTypeWithSinglePlaceholderAndNoExtraTexts(source) && isTemplateLiteralTypeWithSinglePlaceholderAndNoExtraTexts(target)) {

```

Comment:
I'm still considering if this shouldn't have some extra rules on top of this to be completely safe

Reply:
I'm not understanding why this needs to be here. After reverting #55371 the following works just fine:

```ts
declare function bar<T extends string>(x: `${T}`): T;

function baz<U extends string>(x: `${U}`) {
    return bar(x);  // Infers U

}
```

Reply:
Hmm, I confused myself which one of those errored after reverting #55371:
```ts
interface TypeMap {
    a: 'A'
    b: 'B'
}

declare const f: <T extends 'a' | 'b'>(x: `${T}`) => TypeMap[T];

type F1 = <T extends 'a' | 'b'>(x: `${T}`) => TypeMap[T];
const f1: F1 = f;  // Ok

type F2 = <T extends 'a' | 'b'>(x: `${T}`) => TypeMap[`${T}`];
const f2: F2 = f;  // Error, T is not assignable to `${T}`
```

Then I just knew where I could try to improve inference for this as I knew that in here `` `${T}` `` was being inferred for `T` based on `` `${T}` `` source and `` `${T}` `` target as that originally created an issue further down the road.

i agree, this PR is completely redundant 🤦‍♂️ this is what I get for jumping too quickly between work, this, and the kids